src/riak_kv_wm_utils.erl

```diff
@@ -497,6 +499,36 @@ assert_type_and_table_name_same(BucketType1, #ddl_v1{table = BucketType2}) ->
         "    bucket type was: " ++ binary_to_list(BucketType1) ++ "\n"
         "    table name was:  " ++ binary_to_list(BucketType2) ++ "\n"}).
 
+%% @doc Verify that the primary key has three components
+%%      and the first element is a quantum
+assert_partition_key_length(#ddl_v1{partition_key = {key_v1, Key}}) when length(Key) == 3 ->
+    assert_third_param_is_quantum(lists:nth(3, Key));
+assert_partition_key_length(_DDL) ->
+    throw({error, {primary_key, "Primary key is too short"}}).
+
+%% @doc Verify that the first element of the primary key is a quantum
+assert_third_param_is_quantum(#hash_fn_v1{mod=riak_ql_quanta, fn=quantum}) ->
+    ok;
+assert_third_param_is_quantum(_KeyComponent) ->
+    throw({error, {primary_key, "Third element of primary key must be a quantum"}}).
+
+%% @doc Verify primary key and local partition have the same elements
+assert_primary_and_local_keys_match(#ddl_v1{partition_key = #key_v1{ast = Primary}, local_key = #key_v1{ast = Local}}) ->
+    PrimaryList = lists:sort([query_field_name(F) || F <- Primary]),
+    LocalList = lists:sort([query_field_name(F) || F <- Local]),
```

Comment:
Is the order of the fields important, in both keys? Isn't equality of `PrimaryList` and `LocalList` a required and sufficient condition for the test condition to be met?